addons/web/static/src/search/search_bar/search_bar.js

```diff
@@ -98,17 +108,39 @@ export class SearchBar extends Component {
             return;
         }
 
-        for (const searchItem of this.searchItems) {
+        // move properties under their related field
+        const searchItems = [...this.searchItems];
+        searchItems.sort((a, b) => (a.propertyItemId || a.id) - (b.propertyItemId || b.id));
+
+        for (const searchItem of searchItems) {
             const field = this.fields[searchItem.fieldName];
-            const type = field.type === "reference" ? "char" : field.type;
+            let type = field.type === "reference" ? "char" : field.type;
+
+            if (type === "properties" && searchItem.type === "field_property") {
+                // take the definition type instead
+                type = searchItem.definition.type
+
+                if (!this.state.expanded.includes(searchItem.propertyItemId)) {
+                    // properties field not unfolded, do not show the option
+                    continue;
+                }
+            }
+
             /** @todo do something with respect to localization (rtl) */
-            const preposition = this.env._t(["date", "datetime"].includes(type) ? "at" : "for");
+            let preposition = this.env._t(["date", "datetime"].includes(type) ? "at" : "for");
+
+            if (field.type === "properties" && (searchItem.type === "field" || FOLDABLE_TYPES.includes(type))) {
```

Comment:
Hmm still not a fan.
Maybe we could avoid setting the preposition for those in the first place and include everything in the previous condition?


So let preposition = null;
if (!complicated stuff) { set preposition }